kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/KafkaProxyIngressStatusFactory.java

```diff
@@ -13,21 +13,32 @@
 import io.kroxylicious.kubernetes.api.common.Condition;
 import io.kroxylicious.kubernetes.api.v1alpha1.KafkaProxyIngress;
 import io.kroxylicious.kubernetes.api.v1alpha1.KafkaProxyIngressBuilder;
+import io.kroxylicious.kubernetes.api.v1alpha1.KafkaProxyIngressFluent;
 import io.kroxylicious.kubernetes.api.v1alpha1.KafkaProxyIngressStatus;
+import io.kroxylicious.kubernetes.operator.checksum.MetadataChecksumGenerator;
 
 public class KafkaProxyIngressStatusFactory extends StatusFactory<KafkaProxyIngress> {
+
     public KafkaProxyIngressStatusFactory(Clock clock) {
         super(clock);
     }
 
     private KafkaProxyIngress ingressStatusPatch(KafkaProxyIngress observedIngress,
-                                                 Condition condition) {
+                                                 Condition condition,
+                                                 String checksum) {
+
         // @formatter:off
-        return new KafkaProxyIngressBuilder()
+        KafkaProxyIngressFluent<KafkaProxyIngressBuilder>.MetadataNested<KafkaProxyIngressBuilder> metadataBuilder = new KafkaProxyIngressBuilder()
                 .withNewMetadata()
                     .withUid(ResourcesUtil.uid(observedIngress))
                     .withName(ResourcesUtil.name(observedIngress))
-                    .withNamespace(ResourcesUtil.namespace(observedIngress))
+                    .withNamespace(ResourcesUtil.namespace(observedIngress));
+        if (!checksum.isBlank()) {
```

Comment:
feels a bit asymmetrical how we have `String NO_CHECKSUM_SPECIFIED = ""` but test for blankness here, maybe comparing to the constant would make it more obvious we're using blank as a special value?

Reply:
Hmm, thats not something I had thought about. I was hoping that `NO_CHECKSUM_SPECIFIED` would be a temporary thing until I could remove the non-checksum overloads, but I think this check is needed in general so your probably right we should test for not equal to the `NO_CHECKSUM_SPECIFIED` constant.